lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
A solution would be to use `"#{puppet_dir}/config"` instead of `"$puppetdir/config"`